### PR TITLE
Fix GPU test failures: PCA

### PR DESCRIPTION
**Pull request by inteldimitrius**
# Description
Fix PCA test failures: 
- Wrong ASSERT(has_mutable_data()) for this kernel no need to have data being mutable.
- set_variances() to result. 
- Wrong order for sycl::event and wait_and_throw() before setting.

### CODE REVIEW

**Comment by ethanglaser** [comment]
/intelci: run
set run_bazel_testing_pvc=1

**Comment by napetrov** [comment]
@mergify backport rls/2024.0.0-rls

**Comment by ethanglaser** [comment]
http://intel-ci.intel.com/ee5d4b94-b9aa-f18c-8a7f-a4bf010d0e2e

**Comment by ethanglaser** [comment]
> http://intel-ci.intel.com/ee5d4b94-b9aa-f18c-8a7f-a4bf010d0e2e

PCA PVC bazel fails are resolved

File: cpp/oneapi/dal/algo/pca/backend/gpu/train_kernel_precomputed_impl_dpc.cpp [hunk 1]
```diff
@@ -63,7 +63,6 @@ auto compute_eigenvectors_on_host(sycl::queue& q,
                                   std::int64_t component_count,
                                   const dal::backend::event_vector& deps = {}) {
     ONEDAL_PROFILER_TASK(compute_eigenvectors_on_host);
-    ONEDAL_ASSERT(corr.has_mutable_data());
```
Comment (Vika-F) [inline cpp/oneapi/dal/algo/pca/backend/gpu/train_kernel_precomputed_impl_dpc.cpp:66]: I guess this have to be changed to `ONEDAL_ASSERT(corr.has_data());`, not just removed.

Comment (inteldimitrius) [reply cpp/oneapi/dal/algo/pca/backend/gpu/train_kernel_precomputed_impl_dpc.cpp:66]: The following asserts cover this case: 
`ONEDAL_ASSERT(corr.get_dimension(0) == corr.get_dimension(1),
                  "Correlation matrix must be square");
    ONEDAL_ASSERT(corr.get_dimension(0) > 0);`
But I can add separate one with `.has_data()`. What is your opinion?

**Pull request merged**